File: power_set.cpp
//{ Driver Code Starts
#include<bits/stdc++.h>
using namespace std;

// } Driver Code Ends

class Solution{
	public:
		vector<string> AllPossibleStrings(string s){
		    // Code here
		    vector<string> ans;
		    for(int i = 1; i < (1 << s.size()); ++i) {
		        string t;
		        for(int j = 0; j < s.size(); ++j) {
		            if ((i & (1 << j)) != 0) {
		                t += s[j];
		            }
		        }
		        ans.push_back(t);
		    }
		    sort(ans.begin(), ans.end());
		    return ans;
		}
};

//{ Driver Code Starts.
int main(){
	int tc;
	cin >> tc;
	while(tc--){
		string s;
		cin >> s;
		Solution ob;
		vector<string> res = ob.AllPossibleStrings(s);
		for(auto i : res)
			cout << i <<" ";
		cout << "\n";

	}
	return 0;
}
// } Driver Code Ends

File: recursion/power_set.py
# User function Template for python3


class Solution:
    def AllPossibleStrings(self, s):
        # Code here
        ans = []
        s = [c for c in s]
        n = len(s)
        curr = []

        def f(k):
            if k == n:
                ans.append("".join(curr))
                return
            curr.append(s[k])
            f(k + 1)
            curr.pop()
            f(k + 1)

        f(0)
        ans.sort()
        return ans[1:]


# {
# Driver Code Starts

# Initial Template for Python 3

if __name__ == "__main__":
    T = int(input())
    for i in range(T):
        s = input()
        ob = Solution()
        ans = ob.AllPossibleStrings(s)
        for i in ans:
            print(i, end=" ")
        print()
# } Driver Code Ends
